File: customer-address-jpa/src/main/java/com/izertis/example/service/impl/mappers/EventsMapper.java
package com.izertis.example.service.impl.mappers;

import com.izertis.example.events.dtos.Address;
import com.izertis.example.events.dtos.CustomerDeletedEvent;
import com.izertis.example.events.dtos.CustomerEvent;
import com.izertis.example.events.dtos.PaymentMethod;
import com.izertis.example.domain.Customer;
import org.mapstruct.Mapper;
import org.mapstruct.factory.Mappers;

@Mapper(uses = { BaseMapper.class })
public interface EventsMapper {

    EventsMapper INSTANCE = Mappers.getMapper(EventsMapper.class);

    Address asAddress(com.izertis.example.domain.Address address);

    PaymentMethod asPaymentMethod(
            com.izertis.example.domain.PaymentMethod paymentMethod);

    CustomerEvent asCustomerEvent(
            Customer customer);

    CustomerDeletedEvent asCustomerEvent(Long id);

}

File: kustomer-address-jpa/src/main/kotlin/com/izertis/example/service/impl/mappers/EventsMapper.kt
package com.izertis.example.service.impl.mappers

import com.izertis.example.domain.Address
import com.izertis.example.domain.Customer
import com.izertis.example.domain.PaymentMethod
import org.mapstruct.Mapper
import org.mapstruct.factory.Mappers

@Mapper(uses = [BaseMapper::class])
interface EventsMapper {

  companion object {
    val INSTANCE: EventsMapper = Mappers.getMapper(EventsMapper::class.java)
  }

  fun asAddress(address: Address): com.izertis.example.events.dtos.Address

  fun asPaymentMethod(paymentMethod: PaymentMethod): com.izertis.example.events.dtos.PaymentMethod

  fun asCustomerEvent(customer: Customer): com.izertis.example.events.dtos.CustomerEvent

  fun asCustomerEvent(id: Long): com.izertis.example.events.dtos.CustomerEvent {
    return com.izertis.example.events.dtos.CustomerEvent().apply { this.id = id }
  }
}
